components/driver/ledc.c

```diff
@@ -638,6 +638,7 @@ esp_err_t ledc_channel_config(const ledc_channel_config_t *ledc_conf)
             );
     /*set LEDC signal in gpio matrix*/
     gpio_hal_iomux_func_sel(GPIO_PIN_MUX_REG[gpio_num], PIN_FUNC_GPIO);
+    gpio_set_level(gpio_num, output_invert);
```

Comment:
This makes sense.  You should set the pin H/L value before driving the pin as an output.  I'm guessing esp_rom_gpio_connect_out_signal() also sets pin H/L, but by that point you would see a glitch on the pin without this fix.  I'm done the same thing in other hardware drivers I've written.

Reply:
I should also mention esp_rom_gpio_connect_out_signal() is in the ESP32 ROM file, so I didn't find a source file.